rsconnect/pyproject.py

```diff
@@ -16,6 +16,23 @@
     import toml as tomllib  # type: ignore[no-redef]
 
 
+def detect_python_version_requirement(directory: typing.Union[str, pathlib.Path]) -> typing.Optional[str]:
```

Comment:
This is more or less the entry point for the whole version requirement detectiong process, based on the previously merged metadata parsing functions.